apps/voting/contracts/Voting.sol

```diff
@@ -394,17 +496,85 @@ contract Voting is IForwarder, AragonApp {
     * @dev Internal function to check if a voter can participate on a vote. It assumes the queried vote exists.
     * @return True if the given voter can participate a certain vote, false otherwise
     */
-    function _canVote(uint256 _voteId, address _voter) internal view returns (bool) {
-        Vote storage vote_ = votes[_voteId];
+    function _canVote(Vote storage vote_, address _voter) internal view returns (bool) {
         return _isVoteOpen(vote_) && token.balanceOfAt(_voter, vote_.snapshotBlock) > 0;
     }
 
     /**
-    * @dev Internal function to check if a vote is still open
+    * @dev Internal function to check if a representative can vote on behalf of a voter on a certain vote. It assumes the queried vote exists.
+    * @return True if the given representative can vote on behalf of a voter on a certain vote, false otherwise
+    */
+    function _canVoteOnBehalfOf(Vote storage vote_, address _voter, address _representative) internal view returns (bool) {
+        return _canVote(vote_, _voter) &&
+               _isRepresentativeOf(_voter, _representative) &&
+               _hasNotVotedYet(vote_, _voter) &&
+               !_withinOverruleWindow(vote_);
```

Comment:
Nitpick: a small probabilistic optimization here is to reorder the logic.

I imagine most of the time, representative voters will attempt to vote within the time boundaries, and with correct arguments (but as mentioned above in `voteOnBehalfOfMany()`, may have some assumptions move underneath them). Hence:

- `_hasNotVotedYet()` is the most likely to fail, because a voter may have already voted themselves, without the representative knowing
- `_isRepresentativeOf()`, `_withinOverruleWindow()`, and `_canVote()` are mostly error checks to make sure that the representative does, indeed, have the ability to vote.

However, as mentioned in `voteOnBehalfOfMany()`, we may want to split off `_withinOverruleWindow()` and `_canVote()` into their own error check.